usr/share/rear/lib/layout-functions.sh

```diff
@@ -530,10 +530,7 @@ get_disk_size() {
 
     local block_size=$(get_block_size ${disk_name%/*})
 
-    [ -r /sys/block/$disk_name/size ]
-    BugIfError "Could not determine size of disk $disk_name, please file a bug."
-
-    local nr_blocks=$( < /sys/block/$disk_name/size)
+    local nr_blocks=$(retry_command "cat /sys/block/$disk_name/size")
```

Comment:
Taking your example: 
```
retry_command ip a \| grep -q inet6
retry_command test -b /dev/mapper/foobar
```
If command `retry_command ip a \| grep -q inet6` fails after REAR_MAX_RETRIES you will get message _"Could not successfully finish command 'ip a | grep -q inet6'"_ and Error () will end ReaR execution. I don't know what is wrong about that ...